### ISSUE TITLE
Aliases for Darwin Core concept names

### ISSUE THREAD

**Issue by eightysteele** [VertNet/Darwin-Core-Engine#22]
There are 2378 bytes in the 156 Darwin Core concept names. The storage for each entity in the datastore includes this overhead. So for 85 million VertNet records that's 267.41 GB of storage required for concept names alone! 

Put another way, at $0.15 cents per GB per month, that's $40 per month, $481 per year, and $1444 for the three year project to store  concept names.

Workaround? Simple. @tucotuco: Can you create a dictionary mapping between Darwin Core concept names and an alias name less than or equal to 6 bytes? Then in the worst case we're looking at a 74 GB overhead (a 28% savings). 

For example:

```
DWC_ALIAS = dict(
    decimallongitude='lon',
    decimallongitude='lat',
    coordinateuncertaintyinmeters='cuim')
```

The idea is to make the alias somewhat human readable if possible so that in the data viewer console you can understand what they are. With this dictionary we can store aliases in the datastore but query using full concept names.

**Comment by tucotuco** [comment]
Here is a proposal for the shortest possible recognizable codes: https://gist.github.com/1130903 

Average is 2.54 characters per code, total is 397 characters for codes (17% of original name lengths). For Unicode though, each character is 2 bytes, so it looks like storage for codes alone would be about ~63G (~$10.68) if every record had every term. Note: Storage pricing is actually $0.17 per GB per month, not $0.15. My best estimate is that, on average, 68 terms will have data (150 characters, 300 bytes per record, 23.7G total, $4.03 per month) and the average record will have about 1K characters (2k bytes per record, 316.6G total, $53.83). So we're looking at about $58 per month minimum storage requirements.

**Comment by eightysteele** [comment]
LGTM. For readability of the issue and the aliases, I edited your comment and moved them over to a gist: https://gist.github.com/1130903

Let's keep the issue open a bit so that others have a chance to review.

>  For Unicode though, each character is 2 bytes

That's what I thought too, but I checked and it turns out that UTF-8 is a multibyte (1-4) character encoding, so characters whose value is less than 128 (i.e. ASCII) are actually encoded as themselves in one byte (the high-order bit will always be zero).

>  Storage pricing is actually $0.17 per GB per month, not $0.15

Yep, good catch.
